troposphere/static/js/components/admin/ResourceRequest.react.js

```diff
@@ -12,18 +15,202 @@ define(function (require) {
       request: React.PropTypes.instanceOf(Backbone.Model).isRequired
     },
 
+    getInitialState: function(){
+      return {
+        AUSearch: "",
+        delta: 525600,
+        expire: true,
+        response: "",
+        canSubmit: false,
+        newAllocationId: null,
+        displayAdmin: false
+      };
+    },
+
+    handleDisplayChange: function (event){
+      this.setState({displayAdmin: !this.state.displayAdmin});
+    },
+
+    handleResponseChange: function (event) {
+      var response = event.target.value;
+      if (response) this.setState({response: response});
+    },
+
+    handleQuotaChange: function (event) {
+      var quota = event.target.value;
+      if (quota) this.setState({quota: quota});
+    },
+
+    handleAllocationChange: function (event) {
+      var allocation = event.target.value;
+      if (allocation) this.setState({allocation: allocation});
+    },
+
+    handleApproval: function(e){
+        e.preventDefault();
+        var resourceRequest = this.props.request,
+          quotaToSend = parseInt(this.state.quota) || parseInt(resourceRequest.get('current_quota')),
+          allocationToSend = stores.AllocationStore.findWhere({"threshold": parseInt(this.state.AUSearch) * 60, "delta": this.state.delta}).models[0].get('id');
```

Comment:
in theory, you could `.shift()` the first model off this collection and then get the id if it's not null/undefined. I assume that a user is always going to have an allocation, so this is a safe operation...